src/App/App.tsx

```diff
@@ -8,24 +8,33 @@ import Location from "../Location/Location"
 import './App.scss'
 
 import { getCoordinates } from '../helpers/apiCalls'
+import { getAllData } from "../helpers/dataFilter"
+import { DetailsProps } from '../helpers/detailsdefinitions'
 
 
-const App: React.FC = () => {
+const App: React.FC<DetailsProps> = () => {
   const [ searchResults, setSearchResults ] = useState([])
-  const [ query, setQuery ] = useState('')
-	const [ matchDetails, setMatchDetails ] = useState<Array<any>>([])
+  const [ query, setQuery ] = useState<string>()
+  const [ matchDetails, setMatchDetails ] = useState<[number, number, string, string]>()
+  const [ detailsData, setDetailsData ] = useState<DetailsProps>()
+  
 
-  const data = {};
 	const getSearchResults = async (query: string, clearInput: Function) => {
 		setQuery(query)
 		const returnedResults = await getCoordinates(query)
 		setSearchResults(returnedResults)
 		clearInput()
 	}
 	
-	const getMatchDetails = (coordinates: [], locality: string, region: string) => {
-		setMatchDetails([coordinates, locality, region])
-	}
+	const getMatchDetails = (latitude: number, longitude: number, locality: string, region: string) => {
+		setMatchDetails([latitude, longitude, locality, region])
+  }
+
+  const getAllDetailsData = async (lat: number, long: number) => {
```

Comment:
I'm happy it worked out fetching the data here. What was going on with fetching it in Location? I'm curious!

Reply:
I was having issues with the app rerendering somewhere so it was causing loops in Location, even outside of useEffect, lots of trial and error to get it work.